lib/app-service.js

```diff
@@ -5,6 +5,15 @@ var morgan = require("morgan");
 var util = require("util");
 var EventEmitter = require("events").EventEmitter;
 
+// Groups a HTTP log line into stuff before access token, token and stuff after
+//  Example grouping : 
+//      2016-07-26 13:35:41 INFO:IrcBridge ::ffff:127.0.0.1 - - [26/Jul/2016:13:35:41 +0000] "PUT /transactions/146?access_token=
+//      [access token here]
+//       HTTP/1.1" 200 2 "-" "Synapse/0.16.1-r1 (b=master,t=v0.16.1-r1,0870588)"
+var ACCESS_TOKEN_RE = /^(.*\?.*access(?:_|%5[Ff])token=)([0-9a-z]*)([^&]*)$/;
```

Comment:
This regex isn't great. It isn't great because:
 - It fails to get the entire access token if there are any capital letters in it.
 - It fails to get the access token at all if there are subsequent query parameters after the token.

We do not restrict characters on the access token, so we really should be able to handle capital letters. Equally, we should really be able to support the access token appearing as the first / second / last query parameter in the URL.

This feel needlessly complicated for what we're trying to do. Multi-group then join feels clunky when JS has [a replace function for strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace). Why not something like:
```javascript
> f
'something here then access_token=UnPreDictablEString&_tar'
> 
> g
'something here then access_token=UnPreDictablEString'
> 
> f.replace(/access_token=.*?(&|$)/, "access_token=<REDACTED>")
'something here then access_token=<REDACTED>_tar'
> 
> g.replace(/access_token=.*?(&|$)/, "access_token=<REDACTED>")
'something here then access_token=<REDACTED>'
```

This regex does a non-greedy match after the `=` until it reaches a `&` or the end of the line.
This does strip the `&` if it exists (not a big deal), but is overall a lot more durable than this regex is currently.